.github/workflows/test.yml

```diff
@@ -5,6 +5,15 @@ on:
     branches: [master]
   pull_request:
   workflow_dispatch:
+      rose_ref:
+        description: The Rose branch to test against
+        required: false
+      cylc_ref:
+        description: The Cylc branch to test against
+        required: false
+      cylc_rose_ref:
+        description: The Cylc branch to test against
+        required: false
```

Comment:
You could use `default: master` here instead of `${{ github.event.inputs.*_ref || 'master' }}` below

Reply:
This is how I initially tried it, however, the `github.event` "context" is only available when the workflow is run through the `workflow_dispatch` and is not present otherwise meaning that you can't use inputs as a convenient way to store defaults for inputs which is very annoying.

Could provide a "global" `env` section to centralise things.